cabal-install/tests/UnitTests/Distribution/Client/Init/FileCreators.hs

```diff
@@ -0,0 +1,157 @@
+module UnitTests.Distribution.Client.Init.FileCreators (
+  tests
+  ) where
+
+import Distribution.Client.Init.FileCreators
+  ( generateCabalFile )
+
+import Test.Tasty
+import Test.Tasty.Golden (goldenVsString)
+import Test.Tasty.HUnit
+
+import System.FilePath
+  ( (</>) )
+import qualified Data.ByteString.Lazy as BS
+import qualified Data.ByteString.Lazy.Char8 as BS8
+import qualified Data.Set as Set
+
+import Distribution.Client.Init.Types
+  ( InitFlags(..), PackageType(..) )
+import Distribution.Simple.Setup
+  ( Flag(..) )
+
+import Distribution.CabalSpecVersion
+  ( CabalSpecVersion(CabalSpecV2_4) )
+import Distribution.Types.Dependency
+  ( Dependency, mkDependency )
+import Distribution.Types.LibraryName
+  ( LibraryName(LMainLibName) )
+import Distribution.Types.PackageName
+  ( mkPackageName )
+import Distribution.Types.VersionRange
+  ( majorBoundVersion )
+import Distribution.Types.Version
+  ( mkVersion )
+import qualified Distribution.ModuleName as ModuleName
+  ( fromString )
+import qualified Distribution.SPDX as SPDX
+import Language.Haskell.Extension ( Language(..) )
+
+tests :: [TestTree]
+tests = [ testGroup "cabal init goldens"
+          [ checkCabalFileGolden exeFlags "exe-only.cabal"
+          , checkCabalFileGolden libExeAndTestFlags "lib-exe-and-test.cabal"
+          ]
+        ]
+
+checkCabalFileGolden :: InitFlags -> FilePath -> TestTree
+checkCabalFileGolden flags goldenFileName =
+  goldenVsString goldenFileName goldenFilePath generatedCabalFile
+  where
+    inGoldensDir :: FilePath -> FilePath
+    inGoldensDir fn =
+      "tests" </> "UnitTests" </> "Distribution" </> "Client" </> "Init" </>
+      "goldens" </> "generateCabalFile" </>  fn
+
+    goldenFilePath :: FilePath
+    goldenFilePath = inGoldensDir goldenFileName
+
+    generatedCabalFile :: IO BS.ByteString
+    generatedCabalFile = pure . BS8.pack $ generateCabalFile goldenFileName flags
+
+-- ==================================================
+-- Simple exe.
+
+exeFlags :: InitFlags
+exeFlags = InitFlags {
```

Comment:
could this rather be specified in terms of `defaultInitFlags`, or some `commonInitFlags` derived from `defaultInitFlags` (to set package name etc.). Adding new test shouldn't require 40 lines.

Reply:
Done.